integrations/nvidia/tests/test_generator.py

```diff
@@ -116,6 +163,30 @@ def test_run_integration_with_nim_backend(self):
         assert result["replies"]
         assert result["meta"]
 
+    @pytest.mark.integration
+    @pytest.mark.usefixtures("mock_local_models")
+    @pytest.mark.usefixtures("mock_local_chat_completion")
+    def test_run_integration_with_default_model_nim_backend(self):
+        model = None
+        url = "http://localhost:8080/v1"
+        generator = NvidiaGenerator(
+            model=model,
+            api_url=url,
+            api_key=None,
+            model_arguments={
+                "temperature": 0.2,
+            },
+        )
+        with pytest.warns(UserWarning):
```

Comment:
to ensure the expected warning is raised, please name the warning and check its message. for instance -

```
    with pytest.warns(UserWarning) as record:
        ...
    assert len(record) == 1
    assert "Default model set to" in str(record[0].message)
```